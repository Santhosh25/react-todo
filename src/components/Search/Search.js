import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Paper from "@material-ui/core/Paper";
import InputBase from "@material-ui/core/InputBase";
import IconButton from "@material-ui/core/IconButton";
import Icon from "@material-ui/core/Icon";
import { useDispatch, useSelector } from "react-redux";
import actions from "../../actions";

const useStyles = makeStyles((theme) => ({
  root: {
    padding: "2px 4px",
    display: "flex",
    alignItems: "center",
    width: 400,
  },
  input: {
    marginLeft: theme.spacing(1),
    flex: 1,
  },
  iconButton: {
    padding: 10,
  },
  divider: {
    height: 28,
    margin: 4,
  },
}));

const Search = () => {
  const data = useSelector((state) => state.todos.data);
  const classes = useStyles();
  const dispatch = useDispatch();

  return (
    <Paper component="form" className={classes.root}>
      <InputBase
        onChange={(e) => {
          dispatch({
            type: actions.SEARCH,
            payload: {
              data: data,
              text: e.target.value,
            },
          });
        }}
        className={classes.input}
        placeholder="Search Task"
        inputProps={{ "aria-label": "search task" }}
      />
      <IconButton
        type="button"
        className={classes.iconButton}
        aria-label="search"
      >
        <Icon>search</Icon>
      </IconButton>
    </Paper>
  );
};

export default Search;
