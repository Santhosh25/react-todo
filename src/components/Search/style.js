import { Paper } from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";
import styled from "styled-components";

export const useStyles = makeStyles((theme) => ({
  root: {
    boxShadow: "none",
    backgroundColor: "#f0f8ff69",
  },
  input: {
    marginLeft: theme.spacing(1),
    flex: 1,
  },
  iconButton: {
    padding: 10,
  },
  divider: {
    height: 28,
    margin: 4,
  },
}));

export const PaperWrapper = styled(Paper)`
  padding: 2px 4px;
  display: flex;
  align-items: center;
  width: 70%;
  @media (min-width: 1024px) {
    width: 720px;
  }
`;
