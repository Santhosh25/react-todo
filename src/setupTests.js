// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import React from "react";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { configure, shallow, mount, render } from "enzyme";
import "jest-styled-components";
import "@testing-library/jest-dom";
import { Provider } from "react-redux";
import { store } from "./store";

React.useLayoutEffect = React.useEffect;
jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useLayoutEffect: jest.requireActual("react").useEffect,
}));
// const constantDate = new Date('2017-06-13T04:41:20');
// global.Date = class extends Date {
//   constructor(value) {
//     return constantDate;
//   }
// };
configure({ adapter: new Adapter() });

const wrapChildWithThemeProvider = (children) => (
  <Provider store={store}>{children}</Provider>
);

const shallowWithTheme = (children) =>
  shallow(wrapChildWithThemeProvider(children));
const mountWithTheme = (children) =>
  mount(wrapChildWithThemeProvider(children));
const renderWithTheme = (children) =>
  render(wrapChildWithThemeProvider(children));

export { shallowWithTheme, mountWithTheme, renderWithTheme };
