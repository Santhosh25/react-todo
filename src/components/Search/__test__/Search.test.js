import React from "react";

import Search from "../Search";
import { renderWithTheme } from "../../../setupTests";

describe("Search component", () => {
    let wrapper;
    beforeEach(() => {
      wrapper = renderWithTheme(
        <Search />
      );
    });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});