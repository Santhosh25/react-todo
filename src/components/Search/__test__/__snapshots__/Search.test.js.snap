// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders correctly 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": Object {
      "class": "MuiPaper-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "MuiInputBase-root makeStyles-input-2",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "aria-label": "search task",
              "class": "MuiInputBase-input",
              "placeholder": "Search Task",
              "type": "text",
            },
            "children": Array [],
            "name": "input",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "aria-label": undefined,
              "class": undefined,
              "placeholder": undefined,
              "type": undefined,
            },
            "x-attribsPrefix": Object {
              "aria-label": undefined,
              "class": undefined,
              "placeholder": undefined,
              "type": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "aria-label": "search",
            "class": "MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-3",
            "tabindex": "0",
            "type": "button",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "class": "MuiIconButton-label",
              },
              "children": Array [
                Node {
                  "attribs": Object {
                    "aria-hidden": "true",
                    "class": "material-icons MuiIcon-root",
                  },
                  "children": Array [
                    Node {
                      "data": "search",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "span",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                  },
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "button",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "aria-label": undefined,
            "class": undefined,
            "tabindex": undefined,
            "type": undefined,
          },
          "x-attribsPrefix": Object {
            "aria-label": undefined,
            "class": undefined,
            "tabindex": undefined,
            "type": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Node {
        "attribs": Object {
          "aria-label": "search",
          "class": "MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-3",
          "tabindex": "0",
          "type": "button",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "MuiIconButton-label",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "aria-hidden": "true",
                  "class": "material-icons MuiIcon-root",
                },
                "children": Array [
                  Node {
                    "data": "search",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                },
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "button",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "class": "MuiInputBase-root makeStyles-input-2",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "aria-label": "search task",
                "class": "MuiInputBase-input",
                "placeholder": "Search Task",
                "type": "text",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "aria-label": undefined,
                "class": undefined,
                "placeholder": undefined,
                "type": undefined,
              },
              "x-attribsPrefix": Object {
                "aria-label": undefined,
                "class": undefined,
                "placeholder": undefined,
                "type": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "aria-label": undefined,
          "class": undefined,
          "tabindex": undefined,
          "type": undefined,
        },
        "x-attribsPrefix": Object {
          "aria-label": undefined,
          "class": undefined,
          "tabindex": undefined,
          "type": undefined,
        },
      },
    ],
    "name": "form",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
